src/Microsoft.Data/DataFrame.cs

```diff
@@ -301,116 +352,62 @@ public DataFrame Sort(string columnName, bool ascending = true)
+        /// <summary>
+        /// Adds a prefix to the column names
+        /// </summary>
+        public DataFrame AddPrefix(string prefix)
         {
-            DataFrame ret = new DataFrame();
-            if (joinAlgorithm == JoinAlgorithm.Left)
-            {
-                for (int i = 0; i < ColumnCount; i++)
-                {
-                    BaseColumn newColumn = Column(i).Clone();
-                    ret.InsertColumn(ret.ColumnCount, newColumn);
-                }
-                long minLength = Math.Min(RowCount, other.RowCount);
-                PrimitiveColumn<long> mapIndices = new PrimitiveColumn<long>("mapIndices", minLength);
-                for (long i = 0; i < minLength; i++)
-                {
-                    mapIndices[i] = i;
-                }
-                for (int i = 0; i < other.ColumnCount; i++)
-                {
-                    BaseColumn newColumn;
-                    if (other.RowCount < RowCount)
-                    {
-                        newColumn = other.Column(i).Clone(numberOfNullsToAppend: RowCount - other.RowCount);
-                    }
-                    else
-                    {
-                        newColumn = other.Column(i).Clone(mapIndices);
-                    }
-                    SetSuffixForDuplicatedColumnNames(ret, newColumn, leftSuffix, rightSuffix);
-                    ret.InsertColumn(ret.ColumnCount, newColumn);
-                }
-            }
-            else if (joinAlgorithm == JoinAlgorithm.Right)
+            for (int i = 0; i < ColumnCount; i++)
             {
-                long minLength = Math.Min(RowCount, other.RowCount);
-                PrimitiveColumn<long> mapIndices = new PrimitiveColumn<long>("mapIndices", minLength);
-                for (long i = 0; i < minLength; i++)
-                {
-                    mapIndices[i] = i;
-                }
-                for (int i = 0; i < ColumnCount; i++)
-                {
-                    BaseColumn newColumn;
-                    if (RowCount < other.RowCount)
-                    {
-                        newColumn = Column(i).Clone(numberOfNullsToAppend: other.RowCount - RowCount);
-                    }
-                    else
-                    {
-                        newColumn = Column(i).Clone(mapIndices);
-                    }
-                    ret.InsertColumn(ret.ColumnCount, newColumn);
-                }
-                for (int i = 0; i < other.ColumnCount; i++)
-                {
-                    BaseColumn newColumn = other.Column(i).Clone();
-                    SetSuffixForDuplicatedColumnNames(ret, newColumn, leftSuffix, rightSuffix);
-                    ret.InsertColumn(ret.ColumnCount, newColumn);
-                }
+                BaseColumn column = Column(i);
+                _table.SetColumnName(column, prefix + column.Name);
```

Comment:
You're going to need to call `OnColumnsChanged()` after these operations, so the IDataView Schema gets updated properly.